tests/kernel/threads/lifecycle/lifecycle_api/src/test_threads_spawn.c

```diff
@@ -94,3 +94,26 @@ void test_threads_spawn_forever(void)
 	zassert_true(tp2 == 100, NULL);
 	k_thread_abort(tid);
 }
+
+/* Test to validate behavior of multiple calls to k_thread_start() */
+void test_thread_start(void)
+{
+	tp2 = 5;
+
+	k_tid_t tid = k_thread_create(&tdata, tstack, STACK_SIZE,
+			thread_entry_delay, NULL, NULL, NULL,
+			K_HIGHEST_THREAD_PRIO,
+			K_USER, K_FOREVER);
+
+	k_thread_start(tid);
+	k_yield();
```

Comment:
why is this k_yield() call necessary? the child thread is running at highest priority so it should run immediately after the k_thread_start() call right?

Reply:
But the main thread is cooperative thread. Just wanted to demonstrate that a thread, even with higher priority, if it has K_FOREVER delay, it will not be added to ready queue. With k_thread_start(), it will be added to ready queue, and once k_yield() is called, it starts executing. Is my understanding right?

Reply:
Co-operative threads won't get swapped out by the scheduler. They need to voluntarily relinquish the cpu.